### PR TITLE
feat(FEC-10290): upgrade NPM packages

### PR DESCRIPTION
**Pull request by dan-ziv**
### Description of the Changes

:white_check_mark: eslint upgraded to 7.x.x
:white_check_mark: babel upgraded to @babel (7.x.x)
:white_check_mark: webpack upgraded to 4.x.x

Pay attention to the following code changes (done due to Flow upgrade):
* Statics should deleted from interface.
* Declared vars should assigned to constants before export them.

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
 {
   "name": "kaltura-player-js",
   "version": "0.55.0",
-  "main": "dist/kaltura-player.js",
+  "main": "dist/kaltura-ovp-player.js",
```
Comment (Yuvalke) [inline package.json:4]: why did you change the main?

File: package.json [hunk 2]
```diff
@@ -45,66 +46,68 @@
...
     "karma-safari-launcher": "^1.0.0",
     "karma-sourcemap-loader": "^0.3.7",
-    "karma-webpack": "^2.0.2",
-    "lint-staged": "^7.2.0",
-    "mocha": "^3.2.0",
+    "karma-webpack": "^4.0.2",
+    "lint-staged": "^10.2.11",
+    "mocha": "^8.0.1",
     "mocha-cli": "^1.0.1",
-    "playkit-js-env-manager": "https://github.com/kaltura/playkit-js-env-manager.git#v3.3.2",
```
Comment (Yuvalke) [inline package.json:88]: isn't it needed anymore?

File: test/src/common/storage/storage-manager.spec.js [hunk 3]
```diff
@@ -92,10 +75,8 @@ describe('StorageManager', function() {
     });
   });
 
-  it('should set muted to true/false depends on changed volume', function() {
-    sandbox.stub(StorageWrapper, '_testForLocalStorage').callsFake(() => {
-      StorageWrapper._isLocalStorageAvailable = true;
-    });
+  it('should set muted to true/false depends on changed volume', function () {
+    StorageWrapper._testForLocalStorage = () => (StorageWrapper._isLocalStorageAvailable = true);
```
Comment (Yuvalke) [inline test/src/common/storage/storage-manager.spec.js:79]: why did you change the stub?
I think it's better to keep it consistently like [here](https://github.com/kaltura/kaltura-player-js/pull/335/files#diff-3847065bb233ba3c3e34f104575a82fcR180)

File: scripts/job.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+PACKAGE_VERSION=$(cat package.json |
+  grep version |
+  head -1 |
+  awk -F: '{ print $2 }' |
+  sed 's/[",]//g' |
+  tr -d '[[:space:]]')
+
+build() {
+  echo "${reset}${blue}Building player..."
+  NODE_ENV=production PLAYER_TYPE=$PLAYER_TYPE webpack --config webpack.config.js
```
Comment (OrenMe) [inline scripts/job.sh:19]: use `--mode production`\`--mode development`

File: package.json (hunk 1 above)
Comment (dan-ziv) [reply package.json:4]: Because there's no dist file called kaltura-player.js

File: package.json (hunk 2 above)
Comment (dan-ziv) [reply package.json:88]: I don't believe someone using it (unfortunately...) but it was wrongly deleted. thanks

File: test/src/common/storage/storage-manager.spec.js (hunk 3 above)
Comment (dan-ziv) [reply test/src/common/storage/storage-manager.spec.js:79]: There's no _testForLocalStorage member in StorageWrapper so you can't stub it using sinon (at least with the newest version)

File: src/common/cast/remote-player.js [hunk 5]
```diff
@@ -121,21 +121,6 @@ export interface IRemotePlayer {
...
-   */
-  static +Type: string;
-
-  /**
-   * @static
-   * @method
-   * @returns {boolean}
-   * @memberof IRemotePlayer
-   */
-  static isSupported(): boolean;
```
Comment (OrenMe) [inline src/common/cast/remote-player.js:138]: @dan-ziv why remove? because of static? if this is required then define IRemotePlayerStatics and add it there.
